### DISCUSSION TITLE
cropping the image with another trace image

### DISCUSSION THREAD

**Discussion by abilask** [ImageMagick/ImageMagick#4181] [Help]
Hi All,

I am new to ImageMagick,  I need to crop a image using another image. 

I have attached the sample image idea here, it may be already exist or an new idea.

Actually color filling wont work if we want to give some specific patterned color (other than solid color, example wood grains).

If this feature is already available, kindly share me the commands.

![CropTrace](https://user-images.githubusercontent.com/90464818/132873319-9d592ac9-5469-448b-a82a-4d5c6b6e415e.jpg)

**Comment by dlemstra** [comment]
Please only start a single discussions, I have removed the two other ones.

**Comment by snibgo** [comment]
What version of IM, on what platform?

The question is vague. What result do you want? Perhaps the transparency from the first image, with colours from the second. If so, then:
```
magick image2.png image1.png -compose CopyOpacity -composite out.png
```
If this doesn't help, please post actual sample inputs.

**Reply by abilask** [reply]
Hi Snibgo,

The given command did not work.

I have attached the originals (Chasy.jpg & 86.jpg) also the demo sample cropTrace.jpg.

Please try this and let me know the command.

Os: Linux (Ubuntu 20.04)
Version: ImageMagick 6.9.10-23 Q16 x86_64 20190101 


Thanks,

**Comment by abilask** [comment]
![CropTrace](https://user-images.githubusercontent.com/90464818/133647534-cb308611-833d-466d-b80c-8e930028bedb.jpg)
![Chasy](https://user-images.githubusercontent.com/90464818/133647552-135d461e-70ce-4edc-ac06-87017d82ff6b.jpg)
![86](https://user-images.githubusercontent.com/90464818/133647576-599ddd00-c52e-4cf9-a927-cc5e499feb8d.jpg)

**Comment by snibgo** [comment]
The inputs are JPEG, which is horrible. In particular, the mask image should not have been compressed with a JPEG method. If you have better inputs, you should use those.

You haven't said what result you want. Should the result have a transparent background, or opaque white, or what? This gives a transparent background (bash script):

```
magick \
  kitch_trace.jpg \
  -fuzz 10% \
  -fill Black -draw "color 0,0 floodfill" \
  -transparent Black \
  kitch_background.jpg \
  +swap \
  -compose CopyOpacity -composite \
  -trim -resize 600x600 \
  x.png
```
I have trimmed and resized the result to a small size to reduce internet bandwidth problems.
![x](https://user-images.githubusercontent.com/33812368/133651833-01b8c850-5dcd-445f-9021-db9adf446378.png)

**Reply by abilask** [reply]
Thanks Snibgo, Are you recommending PNG instead of JPG or the higher resolution?.

 Is that only out line can be achieved? I was expecting the inner trace also trimmed as a hole.

Opaque white background will be fine.

Note: In my environment "magick" command not working, but "convert" works.

**Comment by snibgo** [comment]
Yes, sorry, you have IM v6, so use "convert" instead of "magick".

If you want a transparent background then a JPEG output is not an option, because JPEG cannot record transparency.

If you want a white background instead of transparent, flatten it over a white background.

For the input, I recommend using images that have not been JPEG compressed. This is because JPEG compression is lossy, which means it changes pixels values. In kitch_trace.jpg, many pixels that look white are not exactly white, so we need "-fuzz 10%".

> I was expecting the inner trace also trimmed as a hole.

Sorry, I don't understand this.

**Comment by snibgo** [comment]
Ah, I think maybe you want this:
```
magick \
  kitch_trace.jpg \
  -fuzz 10% \
  -fill Black -draw "color 0,0 floodfill" \
  -fill Black -draw "color 1275,1650 floodfill" \
  -transparent Black \
  kitch_background.jpg \
  +swap \
  -compose CopyOpacity -composite \
  -background White -compose Over -layers Flatten \
  -trim +repage -resize 600x600 \
  x.png
```
![x](https://user-images.githubusercontent.com/33812368/133719539-355998f9-6f13-4aad-a51f-fb6ebda8a142.png)

**Reply by abilask** [reply]
Thanks Snibgo, I was expecting the inner gap is with white background image as sample marked in pos2.png image also the please refer the expectation marked in other image.

Can we have frame gap, i mean white border / frame?.

![pos2](https://user-images.githubusercontent.com/90464818/133898688-3f8f8344-a0fc-4cb8-ae12-2eacc0f5135e.png)
![pos](https://user-images.githubusercontent.com/90464818/133898732-5498ff10-c3c6-4fc6-b5b3-8afb153dc226.png)

**Comment by snibgo** [comment]
```
magick \
  kitch_trace.jpg \
  -fuzz 10% \
  -fill Black -draw "color 1245,712 floodfill" \
  -transparent White \
  kitch_background.jpg \
  +swap \
  -compose CopyOpacity -composite \
  -background White -compose Over -layers Flatten \
  -trim +repage -resize 600x600 \
  -bordercolor White -border 20 \
  x.png
```
![x](https://user-images.githubusercontent.com/33812368/133906048-5284e513-332b-44b2-b431-31d4fda4a832.png)
